### PR TITLE
Rietz/log work

### PR DESCRIPTION
**Pull request by tmrietz**
Add an extensible logging service
  - Write logInsert, logUpdate, logDelete, entityDiff methods
  - Add example service for AwardType to demonstrate how to use
  - Minor table and model changes to support

### CODE REVIEW

**Comment by tmrietz** [comment]
Looks like a couple signature files snuck in here somehow. Will remove and make sure they're getting .gitignored before merging.

File: erp-server/src/main/java/com/ttt/erp/controller/UserAccountController.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
     }
 
     @GetMapping("/{id}")
-    public List<UserAccount> getUserAccount(@PathVariable("id") final Long id) {
+    public UserAccount getUserAccount(@PathVariable("id") final Long id) {
```
Comment (taylor-jones) [inline erp-server/src/main/java/com/ttt/erp/controller/UserAccountController.java:26]: This makes a lot more sense than returning a List!

File: db/ddl.sql [hunk 2]
```diff
@@ -100,13 +100,14 @@ GRANT ALL ON TABLE award TO "tttAdmin";
...
-DROP TABLE IF EXISTS log CASCADE;
+DROP TABLE IF EXISTS log;
 CREATE TABLE log (
   "id" BIGINT NOT NULL PRIMARY KEY,
   "user_account_id" INT NOT NULL REFERENCES user_account(id),
   "controller_class" VARCHAR(255) NULL,
-  "operation" CRUD_OPERATION NOT NULL,
-  "property" VARCHAR(255) NOT NULL,
+  "subject_id" BIGINT NOT NULL,
+  "operation" varchar(50) NULL,
```
Comment (taylor-jones) [inline db/ddl.sql:109]: I like this change. I was wondering how we would handle it with the ENUM. Just using a string is a nice simplification.

**Review by taylor-jones — APPROVED** [review]
Really nice work. I think it looks great.

**Pull request merged**